### DISCUSSION TITLE
Proposal: Improvement for switch expression with independent conditions

### DISCUSSION THREAD

**Discussion by fonMaringer** [dotnet/csharplang#6458] [Language Ideas]
Sometimes you have to initialize some variable with different values based on independent conditions, e.g. form a validation message or an error message. Earlier you have only one way to do so:

```csharp
string a;
if (condition1())
    a = "some value";
else if (condition2())
    a = "another value";
else if (b && c)
    a = "third value";
else if (...)
    ...
else
    a = "default value";
```

Now you can simply use switch expression with simple workaround for that:

```csharp
var a = true switch
{
    true when condition1() => "some value",
    true when condition2() => "another value",
    true when b && c => "third value",
    ...
    _ => "default value",
}
```

I suggest to improve this construction by removing `true` or `true when`:

```csharp
var a = switch
{
    when condition1() => "some value",
    when condition2() => "another value",
    when b && c => "third value",
    ...
    _ => "default value",
}

OR

var a = switch
{
    condition1() => "some value",
    condition2() => "another value",
    b && c => "third value",
    ...
    _ => "default value",
}
```

**Comment by Richiban** [comment]
(Partial) duplicate of https://github.com/dotnet/csharplang/discussions/6441

**Comment by koszeggy** [comment]
Maybe not the expected answer but instead of a `switch` expression you can use the ternary operator:
```cs
var a = condition1() ? "some value"
    : condition2() ? "another value"
    : b && c ? "third value"
    ...
    : "default value";
```

which is actually even more compact than the suggested `switch` expression.

**Reply by jnm2** [reply]
I've liked using this pattern for many years.
